windows-installer/setup.nsi

```diff
@@ -182,8 +188,10 @@ Section "Install"
   ; Delete the zip file after extraction.
   Delete "$OUTDIR\${ZIP_FILE}"
 
+  ; Create a directory to store buffer files.
```

Comment:
We make the folders before we know if they set the paths to something custom or not?

Reply:
What happens if they're set to something custom (i.e. the placeholder is replaced in the config)? Say the default for one of them is defined as `default/dir/path` but in the config, it's set to `custom/dir/path`. This looks like it'll only create `default/dir/path`. Does the custom one get created at another point?

Reply:
When installing windows agent, it pops up an installer which you can customize the installation directory. Are you sayin it will be replacing the $INSTDIR otherwise the default will be "$env:UserProfile"?
Another question: where we gave the default value all those parameter after CreateDirectory, e.x. ${MAIN_INSTDIR}?

Reply:
@jkschulz By the time the installer runs, `$INSTDIR` is already set.

@sophieyfang Sorry, missed this question. The default value is set by the `InstallDir` directive (line 78).